proposals/guards.md

```diff
@@ -0,0 +1,245 @@
+# Guards
+
+* **Type**: Design proposal
+* **Author**: Alejandro Serrano
+* **Contributors**: Mikhail Zarechenskii
+* **Discussion**: [KEEP-371](https://github.com/Kotlin/KEEP/issues/371)
+
```

Comment:
Maybe there should be a link to [KT-13626](https://youtrack.jetbrains.com/issue/KT-13626) somewhere.